### PR TITLE
ssh/tailssh: flesh out env, support non-pty commands

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #3802

### CODE REVIEW

File: ssh/tailssh/tailssh.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ func (srv *server) handleSSH(s ssh.Session) {
 	}
 	action, localUser, ok := evalSSHPolicy(pol, sctx)
 	if ok && action.Message != "" {
-		io.WriteString(s, action.Message)
+		io.WriteString(s.Stderr(), strings.Replace(action.Message, "\n", "\n\r", -1))
```
Comment (maisem) [inline ssh/tailssh/tailssh.go:153]: fine for now, but we might end up adding a bunch of carriage returns when the input is `\r\n` already. consider pulling it out into a different func and adding a comment.

File: ssh/tailssh/tailssh.go [hunk 2]
```diff
@@ -160,62 +160,105 @@ func (srv *server) handleSSH(s ssh.Session) {
...
-		if u.Username != localUser {
-			fmt.Fprintf(s, "can't switch user\n")
+		cmd = exec.Command(loginShell(lu.Uid))
+	} else {
+		if rawCmd := s.RawCommand(); rawCmd != "" {
+			cmd = exec.Command("/usr/bin/env", "su",
+				"-c", rawCmd,
+				localUser)
+			cmd.Dir = lu.HomeDir
+			cmd.Env = append(cmd.Env,
```
Comment (maisem) [inline ssh/tailssh/tailssh.go:186]: pull this out into a func like you did for loginShell

**Review by maisem — APPROVED** [review]

File: ssh/tailssh/tailssh.go (hunk 2 above)
Comment (bradfitz) [reply ssh/tailssh/tailssh.go]: done

File: ssh/tailssh/tailssh.go (hunk 1 above)
Comment (bradfitz) [reply ssh/tailssh/tailssh.go]: tweaked to \r\n for now but until we have half of a plan here, leaving as-is instead of pretending we do

**Pull request merged**